docs/contributing/code_guide.rst

```diff
@@ -279,39 +279,138 @@ to edit the pull request by changing the corresponding branch on your
 PlasmaPy fork on GitHub.  After a pull request is merged into the
 code, you may delete the branch you created for that pull request.
 
-Commit Messages
----------------
-Good commit messages communicate context and intention to other
-developers and to our future selves.  They provide insight into why we
-chose a particular implementation, and help us avoid past mistakes.
 
-Suggestions on `how to write a git commit message
-<https://cbea.ms/git-commit>`_:
+Comments
+========
+
+A well-placed and well-written comment can prevent future frustrations.
+However, comments are not inherently good. As code evolves, an
+unmaintained comment may become outdated, or get separated from the
+section of code that it was meant to describe. Cryptic and obsolete
+comments may end up confusing contributors. In the worst case, an
+unmaintained comment may contain inaccurate or misleading information
+(hence the saying that "a comment is a lie waiting to happen").
+
+* Refactor code to make it more readable, rather than explaining how it
+  works :cite:p:`wilson:2014`.
+
+* Instead of using a comment to define a variable, rename the variable
+  to encode its meaning and intent.  For example, code like:
+
+  .. code-block:: python
+
+     # collision frequency
+     nu = 1e6 * u.s ** -1
+
+  could be changed to:
+
+  .. code-block:: python
+
+     collision_frequency = 1e6 * u.s ** -1  # no comment needed!
+
+* Use comments to communicate information that you wish you knew before
+  starting to work on a particular section of code, including
+  information that took some time to learn.
+
+* Use comments to communicate information that the code cannot,
+  such as why an alternative approach was *not* taken.
+
+* When possible, include enough contextual information that the comment
+  will make sense even if it is displaced from the code it was
+  originally describing.
```

Comment:
I'm not sure how to visualize this point.  Is there a good example.  I think it's a little confusing without more context. (Sorry, I couldn't resist saying that last bit.)

Reply:
I decided to delete this point since there is already another one that says to provide context.